vyper/codegen/expr.py

```diff
@@ -1038,6 +1039,13 @@ def parse_Call(self):
                 if ret is True:
                     arg_lll.typ = InterfaceType(function_name)  # Cast to Correct interface type.
                     return arg_lll
+
+        elif isinstance(self.expr.func, vy_ast.Attribute) and self.expr.func.attr == "pop":
```

Comment:
Why is this and `append` in two different files?

Reply:
since `append` is void, it can only be called in statement position, e.g.
```
self.foo.append(...)
```
and since `pop` returns a value, it can only be used in expression position, e.g.
```
x = self.foo.pop()
```

Reply:
actually `pop()` can also be used in statement position so I added that to `stmt.py` as well